src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -0,0 +1,266 @@
+
+    @Before
+    public void before(TestContext context) {
+
+        vertx = Vertx.vertx();
+
+        this.httpBridge = new HttpBridge();
+        this.httpBridge.setBridgeConfigProperties(this.bridgeConfigProperties);
+
+        this.vertx.deployVerticle(this.httpBridge, context.asyncAssertSuccess());
+    }
+
+    @After
+    public void after(TestContext context) {
+
+        this.vertx.close(context.asyncAssertSuccess());
+    }
+
+    @Test
+    public void sendSimpleMessage(TestContext context) {
+        String topic = "sendSimpleMessage";
+        kafkaCluster.createTopic(topic, 1, 1);
+
+        Async async = context.async();
+
+        String value = "Hi, This is kafka bridge";
+
+        JsonObject json = new JsonObject();
+        json.put("value", value);
+
+        HttpClient client = vertx.createHttpClient();
+
+        client.post(BRIDGE_PORT, BRIDGE_HOST, "/topic/"+topic, response -> {
+            response.bodyHandler(buffer -> {
+                String deliveryStatus = buffer.toJsonObject().getString("status");
+                context.assertEquals("Accepted", deliveryStatus);
+            });
+        }).putHeader("Content-length", String.valueOf(json.toBuffer().length())).write(json.toBuffer()).end();
+
+        Properties config = kafkaCluster.useTo().getConsumerProperties("groupId", null, OffsetResetStrategy.EARLIEST);
+        config.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        config.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+
+        KafkaConsumer<String, String> consumer = KafkaConsumer.create(this.vertx, config);
+        consumer.handler(record -> {
+            context.assertEquals(record.value(), value);
+            log.info("Message consumed topic={} partition={} offset={}, key={}, value={}",
+                    record.topic(), record.partition(), record.offset(), record.key(), record.value());
+            consumer.close();
+            async.complete();
+        });
+
+        consumer.subscribe(topic, done -> {
+            if (!done.succeeded()) {
+                context.fail(done.cause());
+            }
+        });
+    }
+
+    @Test
+    public void sendSimpleMessageToPartition(TestContext context) {
+        String topic = "sendSimpleMessageToPartition";
+
+        kafkaCluster.createTopic(topic, 2, 1);
+
+        Async async = context.async();
+
+        String value = "Hi, This is kafka bridge";
+
+        int partition = 1;
+
+        JsonObject json = new JsonObject();
+        json.put("value", value);
+        json.put("partition", partition);
+
+        HttpClient client = vertx.createHttpClient();
+
+        client.post(BRIDGE_PORT, BRIDGE_HOST, "/topic/"+topic, response -> {
+            response.bodyHandler(buffer -> {
+                String deliveryStatus = buffer.toJsonObject().getString("status");
+                context.assertEquals("Accepted", deliveryStatus);
```

Comment:
As mentioned before the producer should get record metadata with topic, partition and offset where the message was put. Here you should check that the desired partition (specified in the JSON body by the producer) is the partition where the message was put. At same time you should check that the topic is the same and finally, because this is just a single message, that the destination offset is 0.